### PR TITLE
Media: Backfill audio information for existing videos

### LINKED ISSUE #8499: Video Muting: ensure VideoPress compatibility
## Feature Description

Follow up from https://github.com/google/web-stories-wp/issues/7730 . Ensure that this functionality works with videopress. Both testing the functionality and create a PR to patch the issue. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add logic to backfill videos that do not have post meta already set. 

## Relevant Technical Choices

- Fixed issues where videos were pointing to the wrong location in PHP unit tests
- Used human readable state for muted. Sadly true/false and null do not work as null is default value of default. [See](https://github.com/WordPress/wordpress-develop/blob/80ccb3697c4a6f5638498c9789ed2abea1070b98/src/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php#L470)
- Added migration for existing data. 
- Broke processing into processing poster and processing audio. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Upload video file without sound via Media section of the WP CMS. 
2. Open the editor. 
3. Wait for poster to appear. 
4. See mute icon appear. 

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8499

### CODE REVIEW

File: tests/phpunit/tests/Migrations/Add_VideoPress_Poster_Generation_Media_Source.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Add_VideoPress_Poster_Generation_Media_Source extends Test_Case {
 	public function test_migrate() {
 		$video_attachment_id = self::factory()->attachment->create_object(
 			[
-				'file'           => DIR_TESTDATA . '/images/test-video.mp4',
+				'file'           => DIR_TESTDATA . '/uploads/test-video.mp4',
```
Comment (spacedmonkey) [inline tests/phpunit/tests/Migrations/Add_VideoPress_Poster_Generation_Media_Source.php:35]: The video test file lives here.

File: includes/Migrations/Migrate_Meta_Type.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+namespace Google\Web_Stories\Migrations;
+
+use Google\Web_Stories\Media\Media;
+
+/**
+ * Class Add_Media_Source
+ *
+ * @package Google\Web_Stories\Migrations
+ */
+class Migrate_Meta_Type extends Migrate_Base {
```
Comment (spacedmonkey) [inline includes/Migrations/Migrate_Meta_Type.php]: Might need a better name

**Comment by spacedmonkey** [comment]
This PR will be much easier to test when #8517 is merged

**Comment by swissspidy** [comment]
> * Used human readable state for muted. Sadly true/false and null do not work as null is default value of default. [See](https://github.com/WordPress/wordpress-develop/blob/80ccb3697c4a6f5638498c9789ed2abea1070b98/src/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php#L470)

What's the problem with that?

We can easily set the default value to `null`, that's great.

In the editor we check whether `typeof isMuted !== 'boolean'`  and do the backfilling in that case.

If it's a boolean `false` or `true` we know we have set the value already and don't do the backfilling.

No need for turning this into an enum in `Migrate_Meta_Type` or anything.

It also means we can safely remove `Media::filter_default_value_is_muted()` as the backfilling ensures correct data is set.

File: assets/src/edit-story/app/media/local/typedefs.js [hunk 3]
```diff
@@ -83,8 +83,10 @@
  * @property {string} mediaType current media type filter selected
  * @property {string} nextPageToken the page token for the next page
  * @property {string} pageToken page token
- * @property {Array.<number>} processed videos that have been processed
- * @property {Array.<number>} processing videos that haven't been processed
+ * @property {Array.<number>} audioProcessing videos that have been had audio processed
+ * @property {Array.<number>} audioProcessed videos that haven't been had their audio processed
+ * @property {Array.<number>} posterProcessing videos that have been had poster processed
+ * @property {Array.<number>} posterProcessed videos that haven't been had their poster processed
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/local/typedefs.js]:
```suggestion
 * @property {Array.<number>} audioProcessing Videos that have been processed for audio track detection.
 * @property {Array.<number>} audioProcessed Videos that haven't yet been processed for audio track detection.
 * @property {Array.<number>} posterProcessing Videos that have been processed for poster generation.
 * @property {Array.<number>} posterProcessed Videos that haven't yet been processed for poster generation.
```

File: assets/src/edit-story/app/media/local/typedefs.js [hunk 4]
```diff
@@ -109,9 +111,10 @@
  * @property {string} mediaType current media type filter selected
  * @property {string} nextPageToken the page token for the next page
  * @property {string} pageToken page token
- * @property {Array.<number>} processed ids of videos that have been processed
- * @property {Array.<number>} processing ids of videos that haven't been
- * processed
+ * @property {Array.<number>} audioProcessed ids of videos that have been processed audio
+ * @property {Array.<number>} audioProcessing ids of videos that haven't processed audio
+ * @property {Array.<number>} posterProcessed ids of videos that have been processed poster
+ * @property {Array.<number>} posterProcessing ids of videos that haven't processed poster
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/local/typedefs.js]:
```suggestion
 * @property {Array.<number>} audioProcessed IDs of videos that have been processed for audio track detection.
 * @property {Array.<number>} audioProcessing IDs of videos that haven't yet been processed for audio track detection.
 * @property {Array.<number>} posterProcessed IDs of videos that have been processed for poster generation.
 * @property {Array.<number>} posterProcessing IDs of videos that haven't yet been processed for poster generation.
```

**Comment by spacedmonkey** [comment]
> We can easily set the default value to null, that's great.

I tried over over an hour to getting it working. It doesn't work. It is not possible to get default value to null. If you set the default to null, is simple doesn't allow [see](https://github.com/WordPress/wordpress-develop/blob/80ccb3697c4a6f5638498c9789ed2abea1070b98/src/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php#L479-L481). Please believe me on this one, I [introduced](https://core.trac.wordpress.org/changeset/48402) this to core. To get it work, there would need to changed to core. 

This is a big change and I didn't want to spend 4 hours making this change, but it simple did work any other way.

**Comment by spacedmonkey** [comment]
> It also means we can safely remove Media::filter_default_value_is_muted() as the backfilling ensures correct data is set.

Not sure why you don't like the default value filter. The logic of it has changed in this PR. It still works well,  not sure why it would need to removed.

**Comment by spacedmonkey** [comment]
As there are limitations on default values, on register post meta. I think we should avoid using register meta for rest api. Instead registering own field in the api, gives the control we need here. Here is my prototype. 

https://github.com/google/web-stories-wp/tree/feature/backfill-mute-data-1

File: assets/src/edit-story/app/api/apiProvider.js [hunk 5]
```diff
@@ -206,7 +206,7 @@ function APIProvider({ children }) {
           'alt_text',
           'source_url',
           'media_source',
-          'meta.web_stories_is_muted',
+          'is_muted',
```
Comment (swissspidy) [inline assets/src/edit-story/app/api/apiProvider.js:209]: Also needs to be updated in `edit-story.php`

**Comment by spacedmonkey** [comment]
Noticed the media class is getting a little large now. Moving to new video_muting class was good in this PR. Created a new ticket to follow up on breaking up that class more https://github.com/google/web-stories-wp/issues/8564

File: .phpstorm.meta.php [hunk 6]
```diff
@@ -63,6 +63,7 @@
 			'user_preferences'             => \Google\Web_Stories\User\Preferences::class,
 			'web_stories_block'            => \Google\Web_Stories\Block\Web_Stories_Block::class,
 			'video_captions'               => \Google\Web_Stories\Media\Video_Captions::class,
+			'video_muting'                 => \Google\Web_Stories\Media\Video_Muting::class,
```
Comment (swissspidy) [inline .phpstorm.meta.php:66]: Should we prefix all media services with `media.`?

**Review by swissspidy — APPROVED** [review]
Works like a charm! 🚀

File: .phpstorm.meta.php (hunk 6 above)
Comment (spacedmonkey) [reply .phpstorm.meta.php:66]: Video caption like the line above doesn't have media at the start of it.

Comment (swissspidy) [reply .phpstorm.meta.php:66]: That's why I am asking. Should we start doing this?

Just like the `rest.` prefix for REST controllers.

To keep the list tidy and organized.

**Pull request merged**